driver-core/src/test/unit/com/mongodb/connection/InternalStreamConnectionSpecification.groovy

```diff
@@ -581,39 +557,51 @@ class InternalStreamConnectionSpecification extends Specification {
         int threads = 10
         int numberOfOperations = 10000
         ExecutorService pool = Executors.newFixedThreadPool(threads)
+        ExecutorService streamPool = Executors.newFixedThreadPool(4)
```

Comment:
Without the streamPool I get a StackOverflowException, which makes sense.  But I'm not sure why that doesn't happen on master.